packages/pds/tests/db.test.ts

```diff
@@ -1,5 +1,12 @@
 import { TestNetworkNoAppView } from '@atproto/dev-env'
-import { AccountDb } from '../src/account-manager/db'
+
+// @NOTE Not using an import from "../src" because this results in types from
+// "src" and "dst" being mixed (because of the circular dependency between
+// dev-end and pds). That mixing of types is usually not a problem (as they are
+// considered as two different implementation of the same interfaces by TS),
+// except when `unique symbol` are used, which is the case for
+// `jsonEncodedType` used by DB definitions.
```

Comment:
In that case this is a bit of a sharp edge around `jsonEncodedType`, I could imagine this being difficult to catch if you bump into it.  Is a unique symbol needed as opposed to a global one?

Reply:
A `unique symbol` is basically the same as doing `typeof someSymbolVariable`. The issue here is that we mix symbols from `src` and `dist`, which doesn't work whether they are declared as `unique symbol` or as a variable.

Reply:
Note that we _could_ use a `string` instead of a `symbol` here (e.g. `__encodedType`) but that would mean that the following would be allowed:

```ts
const foo: JsonEncoded<{ foo: number }> = '{"foo":42}'
foo.__encodedType.foo // TS: 42; Runtime: Error
```

The use of a `unique symbol` is the proper TS way of implementing tagged types. The caveat here is only needed because of the circular dependency between `dev-env` and `pds`.

Reply:
Note that I changed the implementation to use an import from `dist`, which does fix the issue.